### ISSUE TITLE
[Design] Support for configuring MinimumHealthyPercent and MaximumPercent

### ISSUE THREAD

**Issue by paragbhingre** [aws/copilot-cli#3427]
### Background ◀️

Today, copilot only support [rolling](https://docs.aws.amazon.com/whitepapers/latest/overview-deployment-options/rolling-deployments.html) deployment with `minimumHealthyPercent:100` and `maximumPercent: 200` where ECS spins up new tasks, wait until they are stable before stopping the old tasks.

However, some users want ECS to stop the existing ones before spinning up new ones, for example [issue/3153](https://github.com/aws/copilot-cli/issues/3153). Under the hood, this is setting `minimumHealthyPercent` to `0`, and `maximumPercent` to `100`.

### Proposal
We propose adding support for the above use case through a new field `deployment` in the manifest:

```
deployment:
    rolling: 'recreate' 
```

This will set `minimumHealthyPercent` and `maximumPercent` to `0/100` respectively.

Note that `deployment` is an optional fields in the manifest. If customers doesn’t set any value to it then they will see default behavior of copilot during deployment, which is `100/200` for `min/max`. This maintains backward compatibility.

### An Example
If we do not set `deployment` parameter with the value as rolling: 'recreate' then copilot default behavior will be applied. The default value of minimumHealthyPercent and maximumPercent will be set i.e. 100 and 200. 
If you have desired task count as 4, minimum healthy percent as 100% and maximum percent of 200%. Then ECS can spin up 4 new tasks before even stopping previous 4 tasks

If we do set `deployment` parameter with the value as rolling: 'recreate' then minimumHealthyPercent will be set to 0 and maximumPercent to 100. 
If you have desired task count as 1, minimum healthy percent as 0% and maximum percent of 100%. Then ECS can spin up 1 new task only after stopping the previous task. This is the exact use case we are trying solve through this proposal.
